module/zfs/vdev_queue.c

```diff
@@ -723,15 +744,6 @@ vdev_queue_aggregate(vdev_queue_t *vq, zio_t *zio)
 	while ((dio = zio_walk_parents(aio, &zl)) != NULL) {
 		ASSERT3U(dio->io_type, ==, aio->io_type);
 
-		if (dio->io_flags & ZIO_FLAG_NODATA) {
```

Comment:
The comment `Use the dropped lock...` above isn't accurate after this change, can you remove that line.